cmake/llvm-project.cmake

```diff
@@ -27,6 +27,9 @@ set(LLVM_BUILD_EXAMPLES ON CACHE BOOL "")
 set(LLVM_INSTALL_UTILS ON CACHE BOOL "")
 set(LLVM_ENABLE_TERMINFO OFF CACHE BOOL "")
 set(LLVM_ENABLE_ASSERTIONS ON CACHE BOOL "")
+if(MSVC)
+  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -D_SILENCE_NONFLOATING_COMPLEX_DEPRECATION_WARNING" CACHE INTERNAL "")
```

Comment:
What's requiring this? Is there some upstream bug here? Is there a better way to do this, or a more targeted way?

Reply:
According to the C++ ISO spec, §26.2/2, the effect of instantiation of the template `complex` for any type other than `float`, `double`, or `long double` is unspecified. Clang, GNU, and MSVC provide template specializations for those types and provide the template itself. However, the MSVC warns about using the template, not specializations, and mentions that it is deprecated. To silence the warning, one may use `_SILENCE_NONFLOATING_COMPLEX_DEPRECATION_WARNING`.
Here we may safely turn off the warnings because the data type used is a class for representing a floating point number, and the class provides all required operations for the `complex` template to work correctly.

Reply:
Per discussion, we can safely put this here and then try to make it an upstream patch eventually.